src/settings/environment.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+use std::env;
+
+use config::{Source, Value};
+
+pub trait QueryEnvironment {
+    fn get_var(&self, var: &'static str) -> Result<String, std::env::VarError>;
+
+    fn empty(&self) -> Result<bool, failure::Error>;
+}
+
+#[derive(Clone, Debug)]
+pub struct Environment {
+    whitelist: Vec<&'static str>,
```

Comment:
this whitelist means we don't pull the entire environment in to assess things, which is nice.

Reply:
great step towards #867